Frameworks/UIKit/StarboardXaml/LayerCoordinator.cpp

```diff
@@ -0,0 +1,1335 @@
+//******************************************************************************
+//
+// Copyright (c) 2015 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+// clang-format does not like C++/CX
+// clang-format off
+
+#include "LayerCoordinator.h"
+#include "LayerProxy.h"
+
+#include <ErrorHandling.h>
+
+#include <algorithm>
+#include <functional>
+#include <map>
+
+using namespace Platform;
+using namespace UIKit::Private::CoreAnimation;
+using namespace Windows::Foundation;
+using namespace Windows::Storage::Streams;
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Controls;
+using namespace Windows::UI::Xaml::Media;
+using namespace Windows::UI::Xaml::Media::Animation;
+using namespace Windows::UI::Xaml::Media::Imaging;
+
+static const wchar_t* TAG = L"LayerCoordinator";
```

Comment:
>LayerCoordinator [](start = 30, length = 16)

A lot of this came from either CALayerXaml.cpp or XamlCompositor.cpp; sorry it's not easy to diff.